numpy/core/src/umath/loops.c.src

```diff
@@ -1124,15 +1124,17 @@ TIMEDELTA_md_m_multiply(char **args, npy_intp const *dimensions, npy_intp const
     BINARY_LOOP {
         const npy_timedelta in1 = *(npy_timedelta *)ip1;
         const double in2 = *(double *)ip2;
-        if (in1 == NPY_DATETIME_NAT) {
+        if (in1 == NPY_DATETIME_NAT || npy_isnan(in2)) {
             *((npy_timedelta *)op1) = NPY_DATETIME_NAT;
         }
         else {
-            double result = in1 * in2;
-            if (npy_isfinite(result)) {
-                *((npy_timedelta *)op1) = (npy_timedelta)result;
-            }
-            else {
+            /* `nearbyint` avoids warnings (should not matter here, though) */
+            double result = nearbyint(in1 * in2);
```

Comment:
is `nearbyint` similar to python `round`?

Reply:
roughly speaking, yes.  Of course it will round C-style.  It is really the same thing as `np.rint`, except that it should give a warning (which should not really matter.  If it sets a warning it should be the one we set anyway, but...)